### PR TITLE
Get public key workaround

### PR DESCRIPTION
**Pull request by cool-user**

### CODE REVIEW

File: src/getExtendedPublicKey.h [hunk 1]
```diff
@@ -17,7 +17,15 @@ typedef struct {
 	uint8_t chainCode[32];
 	cx_ecfp_public_key_t publicKey;
 	cx_ecfp_256_extended_private_key_t privateKey;
-	// cx_ecfp_private_key_t privateKey;
 } get_ext_pub_key_data_t;
 
+static unsigned char const WIDE  C_cx_Ed25519_Bx[] = {
```
Comment (ppershing) [inline src/getExtendedPublicKey.h]: Aren't these defined somewhere in cx.h ?

File: src/getExtendedPublicKey.c [hunk 2]
```diff
@@ -66,6 +66,34 @@ void initializePath(uint8_t *dataBuffer)
 	VALIDATE_PARAM(data.bip32Path[2] >= HARDENED_BIP32);
 }
 
+static void os_swap_bigint32(unsigned char *bigint)
```
Comment (ppershing) [inline src/getExtendedPublicKey.c]: I guess we might want to extract this into a general and STL-like reverse(begin, end) call ...

File: src/getExtendedPublicKey.c [hunk 3]
```diff
@@ -66,6 +66,34 @@ void initializePath(uint8_t *dataBuffer)
...
+		unsigned char x;
+		x = bigint[i];
+		bigint[i] = bigint[31-i];
+		bigint[31-i] = x;
+	}
+}
+
+// this is a workaround for SDK <= 1.5.4?, we use this instead of the cx_ecfp_generate_pair
+// as these functions do not work with extended private key
+void derivePublicKey()
```
Comment (ppershing) [inline src/getExtendedPublicKey.c]: Later we would need to make these functions to get in/out data from parameters instead of a global state

**Comment by cool-user** [comment]
So cslashm told us that we should not use this solution. There is another workaround.

**Pull request closed without merging**